src/words/environment/WordsSay.java

```diff
@@ -1,22 +1,49 @@
 package words.environment;
 import java.util.LinkedList;
 
+import words.ast.AST;
+import words.ast.ASTValue;
+import words.ast.ASTValue.ValueType;
+import words.exceptions.WordsInvalidTypeException;
+import words.exceptions.WordsProgramException;
+import words.exceptions.WordsRuntimeException;
+
 public class WordsSay extends WordsAction {
-	private String message;
-	// TODO: Should also allow expression
-	
-	public WordsSay(String message) {
-		this.message = message;
+	private AST messageExpression;
+	private String messageValue;
+
+	public WordsSay(AST messageExpression) {
+		this.messageExpression = messageExpression;
+	}
+
+	public WordsSay(String messageValue) {
```

Comment:
I don't think we should allow this form of construction anymore.  This was the old, incorrect way of doing it.  The grammar only allows an expression, so I think we should do the same here.  That means some existing tests will need to be modified, but that's OK and should be easy to change.